prisma/migrations/20240518020747_pending_items/migration.sql

```diff
@@ -0,0 +1,219 @@
+        INTO med_votes FROM "Item" WHERE "userId" = item."userId";
+
+    -- if their median votes are positive, start at 0
+    -- if the median votes are negative, start their post with that many down votes
+    -- basically: if their median post is bad, presume this post is too
+    -- addendum: if they're an anon poster, always start at 0
+    IF med_votes >= 0 OR item."userId" = 27 THEN
+        med_votes := 0;
+    ELSE
+        med_votes := ABS(med_votes);
+    END IF;
+
+    -- there's no great way to set default column values when using json_populate_record
+    -- so we need to only select fields with non-null values that way when func input
+    -- does not include a value, the default value is used instead of null
+    SELECT string_agg(quote_ident(key), ',') INTO select_clause
+    FROM jsonb_object_keys(jsonb_strip_nulls(jitem)) k(key);
+    -- insert the item
+    EXECUTE format($fmt$
+        INSERT INTO "Item" (%s, "weightedDownVotes", freebie)
+        SELECT %1$s, %L, %L
+        FROM jsonb_populate_record(NULL::"Item", %L) RETURNING *
+    $fmt$, select_clause, med_votes, freebie, jitem) INTO item;
+
+    INSERT INTO "ItemForward" ("itemId", "userId", "pct")
+        SELECT item.id, "userId", "pct" FROM jsonb_populate_recordset(NULL::"ItemForward", forward);
+
+    -- Automatically subscribe to one's own posts
+    INSERT INTO "ThreadSubscription" ("itemId", "userId")
+    VALUES (item.id, item."userId");
+
+    -- Automatically subscribe forward recipients to the new post
+    INSERT INTO "ThreadSubscription" ("itemId", "userId")
+        SELECT item.id, "userId" FROM jsonb_populate_recordset(NULL::"ItemForward", forward);
+
+    INSERT INTO "PollOption" ("itemId", "option")
+        SELECT item.id, "option" FROM jsonb_array_elements_text(poll_options) o("option");
+
+    -- if this is a bio
+    IF item.bio THEN
+        UPDATE users SET "bioId" = item.id WHERE id = item."userId";
+    END IF;
+
+    -- record attachments
+    IF upload_ids IS NOT NULL THEN
+        INSERT INTO "ItemUpload" ("itemId", "uploadId")
+            SELECT item.id, * FROM UNNEST(upload_ids) ON CONFLICT DO NOTHING;
+    END IF;
+
+    -- schedule imgproxy job
+    INSERT INTO pgboss.job (name, data, retrylimit, retrybackoff, startafter)
+    VALUES ('imgproxy', jsonb_build_object('id', item.id), 21, true, now() + interval '5 seconds');
+
+    -- if item is pending, user pays missing sats later.
+    -- all remaining queries will run when invoice was paid and we update the item status.
+    IF item."status" = 'PENDING'::"Status" THEN
+        -- here, we immediately deduct as many of the sats that are required for the payment
+        -- to effectively "lock" them for it. the remainder will be paid via invoice.
+        -- if the payment fails, we release the locked sats by adding them to the balance again.
+        UPDATE users SET msats = GREATEST(msats - cost_msats - item.boost, 0) WHERE id = item."userId";
```

Comment:
This pr is a big lift even without this *partial payment* change, but you might have this working well enough it's okay to add. But if it is overwhelming getting this extra done, I'd recommend postponing the *partial payments* until the *primary* objective of this PR is pristine. 

It's usually a lot easier to add extras when the foundation is settled.

Reply:
Mhh yes, but so far, I think it's only a matter of adding them back in case of failure which I have to handle anyway. So let's see

Reply:
Ok, since deducting user balance is not idempotent and even though LND subscriptions don't trigger on invoice expiration (so idempotency shouldn't be necessary), I removed partial payments in 2c1807d2 to be sure.